pkg/cmd/workflow.go

```diff
@@ -0,0 +1,364 @@
+
+	if err != nil {
+		return err
+	}
+
+	workflowName, err := getWorkflowName(args)
+
+	if err != nil {
+		return err
+	}
+
+	Dialog.Info("Replacing workflow " + workflowName)
+
+	workflow, werr := Client.GetWorkflow(workflowName)
+
+	if werr != nil {
+		return werr
+	}
+
+	revision, rerr := Client.CreateRevision(workflowName, file)
+
+	if rerr != nil {
+		return rerr
+	}
+
+	wr := model.NewWorkflowRevision(workflow.Workflow, revision.Revision)
+
+	wr.Output(Config)
+
+	Dialog.Infof(`Successfully updated workflow %v
+			
+Updated configuration is visible at %v`,
+		workflowName,
+		format.GuiLink(Config, "/workflow/%v", workflowName),
+	)
+
+	return nil
+}
+
+func newReplaceWorkflowCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "replace [workflow name]",
+		Short: "Replace the configuration of a relay workflow",
+		Args:  cobra.MaximumNArgs(1),
+		RunE:  doReplaceWorkflow,
+	}
+
+	cmd.Flags().StringP("file", "f", "", "Path to relay workflow file.")
+
+	return cmd
+}
+
+func doDeleteWorkflow(cmd *cobra.Command, args []string) error {
+	workflowName, nerr := getWorkflowName(args)
+
+	if nerr != nil {
+		return nerr
+	}
+
+	proceed, cerr := util.Confirm("Are you sure you want to delete this workflow?", Config)
+
+	if cerr != nil {
+		return cerr
+	}
+
+	if !proceed {
+		return nil
+	}
+
+	Dialog.Progress("Deleting workflow...")
+
+	_, err := Client.DeleteWorkflow(workflowName)
+
+	if err != nil {
+		return err
+	}
+
+	Dialog.Info("Workflow successfully deleted")
+
+	return nil
+}
+
+func newDeleteWorkflowCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete [workflow name]",
+		Short: "Delete a relay workflow",
+		Args:  cobra.MaximumNArgs(1),
+		RunE:  doDeleteWorkflow,
+	}
+
+	return cmd
+}
+
+func parseParameter(str string) (key, value string) {
+	idx := strings.Index(str, "=")
+
+	// TODO: This behavior will basically silently discard parameters that are
+	// in the wrong format. Should this, like, panic perhaps? Or notify the user that their
+	// parameters are specified incorrectly?
+	if idx < 0 {
+		return
+	}
+
+	key = str[0:idx]
+	value = str[idx+1:]
+	return
+}
+
+func parseParameters(strs []string) map[string]string {
+	res := make(map[string]string)
+
+	for _, str := range strs {
+		key, val := parseParameter(str)
+
+		// value of empty string could, indeed, be a valid parameter.
+		if key != "" {
+			res[key] = val
+		}
+	}
+
+	return res
+}
+
+func doRunWorkflow(cmd *cobra.Command, args []string) error {
+	params, err := cmd.Flags().GetStringArray("parameter")
+
+	if err != nil {
+		panic(err)
+	}
+
+	// TODO: Same here as above. Could really DRY all this up.
+	name, err := getWorkflowName(args)
+
+	if err != nil {
+		return err
+	}
+
+	Dialog.Progress("Starting your workflow...")
+
+	resp, err := Client.RunWorkflow(name, parseParameters(params))
+
+	if err != nil {
+		// TODO: This error should be translated for the user. Right now it just
+		// says whatever the service says.
```

Comment:
I think we want to pipe through errors from the service, that's how we get validation messages and other good stuff. We could add some nice cli-specific layers on top of that to be extra helpful I suppose.

Reply:
+1 to just sending it back from the service, especially if that means you see the same error text in CLI and GUI - makes documenting common problems tons easier

Reply:
Sounds good. That's the default behavior right now. I'll remove this commentary.